afs/models.py

```diff
@@ -196,39 +209,95 @@ def upload_model(
         # Fetch tags apm_node
         pai_data_dir = os.getenv("PAI_DATA_DIR", None)
         if pai_data_dir:
-            pai_data_dir = json.loads(pai_data_dir)
-            if "data" in pai_data_dir:
-                data = pai_data_dir["data"]
-                if "machineIdList" in data:
-                    machineIdList = data.get('machineIdList', [None]).pop(0)
-                    if machineIdList:
-                        tags.update({"apm_node": str(machineIdList)})
+            try:
+                pai_data_dir = json.loads(pai_data_dir)
+                if "data" in pai_data_dir and "type" in pai_data_dir:
+                    data = pai_data_dir["data"]
+                    firehose_type = pai_data_dir["type"]
+
+                    if "machineIdList" in data and firehose_type == "apm-firehose":
+                        machineIdList = data.get("machineIdList", [None]).pop(0)
+
+                        if machineIdList:
+                            tags.update({"apm_node": str(machineIdList)})
+            except Exception as e:
+                print(
+                    f"PAI_DATA_DIR value is not valid json format for apm_node. Exception:{e}, Value: {pai_data_dir}"
+                )
 
         # evaluation result
         evaluation_result.update(extra_evaluation)
         data = dict(
             tags=json.dumps(tags), evaluation_result=json.dumps(evaluation_result)
         )
-        files = {"model": model_file}
-
+        # model name
         if model_name:
             self._naming_rule(model_name)
             data.update({"name": model_name})
 
         extra_paths = [self.repo_id, self.sub_entity_uri]
-        resp = self._create(
-            data=data, files=files, extra_paths=extra_paths, form="data"
-        )
+        # Load model size < 300 MB
+        file_size = os.path.getsize(model_path)
+        if file_size < (300 * 1024 * 1024) and not blob_mode:
```

Comment:
I don't think separate to two different implementations would be a good choice. Why not just one implementation(upload to blob directly)?

Reply:
This implementation will be applied to all upload model methods. However, I want to implement this method in stages. The way to retain the previous version at this stage is to maintain availability, as this change may affect the availability of other features on the server.